src/js/components/header.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { toggleWidget, showSettings, hideSettings } from 'actions/app-state-actions'
+
+
+export class HeaderComponent extends Component {
+    constructor(props) {
+        super(props);
+        this.actions = this.props.actions;
+
+        this.showSettings = this.showSettings.bind(this);
+        this.hideSettings = this.hideSettings.bind(this);
+    }
+
+    showSettings(e) {
+        e.stopPropagation();
+        this.actions.showSettings();
+    }
+
+    hideSettings(e) {
+        e.stopPropagation();
+        this.actions.hideSettings();
+    }
+
+    render() {
+        let {settingsEnabled, settingsVisible, widgetOpened} = this.props.appState;
+        let {settingsHeaderText, headerText} = this.props.ui.text;
+
+        let unreadTimestamp = this.props.appState.messageReadTimestamp;
+
+        let unreadMessagesCount = unreadTimestamp <= 0 ? 0 : this.props.conversation.messages.reduce((count, message) => {
+            if (message.role !== 'appUser' && unreadTimestamp <= message.received) {
+                return count + 1;
+            }
+            return count;
+        }, 0);
+
+        let unreadBadge = !settingsVisible && unreadMessagesCount > 0 ? (
```

Comment:
style nit, don't need to wrap all jsx in (). Eg this could be:

```
        let unreadBadge = !settingsVisible && unreadMessagesCount > 0 ?
            <div id="sk-badge">
                { unreadMessagesCount }
            </div> : null;
```